psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c

```diff
@@ -0,0 +1,234 @@
+            temperature_values[7-j]=1;
+        }else{
+            temperature_values[7-j]=0;
+        }
+    }
+    decoded_temperature_data = (uint8_t)((temperature_values[7]<<7) + (temperature_values[6]<<6) + (temperature_values[5]<<5) + (temperature_values[4]<<4) + (temperature_values[3] <<3) + (temperature_values[2] <<2) + (temperature_values[1]<<1) + temperature_values[0]);
+    decoded_humidity_data=(uint8_t)((humidity_values[7]<<7) + (humidity_values[6]<<6) + (humidity_values[5]<<5) + (humidity_values[4]<<4) + (humidity_values[3] <<3) + (humidity_values[2] <<2) + (humidity_values[1]<<1) + humidity_values[0]);
+    
+    //Now the data is decoded
+    decode_flag = 1;
+    //From here we will have the acctual values. We can call functions from
+    //here or send it out depending on what we decide to go with 
+}    
+
+//Name: Delay_Timer_ISR_Handler
+//Summary:
+//    Interrupt Service Routine. Check the Delay_Timer status and clear the interrupt
+//    change the Count and Executed status
+
+_CY_ISR(Delay_Timer_ISR_Handler){
+    
+    if(Delay_Timer_ReadPeriod() == DELAY_ACQUISITION){
+        count=2;                  
+        executed=0;             
+    }else if(Delay_Timer_ReadPeriod() == DELAY_PROCESSING){ 
+        decode_flag=1;
+    }
+    
+    //Write 1 to the Contro_Reg_Delay_Timer to disable the Timer function 
+    Control_Reg_Delay_Timer_Write(1);
+    //Read the status register of Delay_Timer to clear interrupt
+    Delay_Timer_ReadStatusRegister();
+    //Clear the Delay_Timer_ISR interrupt
+    Delay_Timer_ISR_ClearPending();
+}    
+
+_CY_ISR(DHT_Pin_ISR_Handler){
+    //Clear the interrupt request of the ISR DHT_Pin_ISR 
+    DHT_Pin_ISR_ClearPending();
+    //Clear the interrupt request of the pin DHT_Pin
+    DHT_Pin_ClearInterrupt();
+    //Read the counter values of Data_Timer for processing
+    countvalues_datatimer[risingedge_counts]=Data_Timer_ReadCounter();
+    risingedge_counts++;
+}
+
+void dht_init(){
+    check = 0;
+    decode_flag = 0;
+    
+    //Starts the data timer and delay timer components
+    Data_Timer_Start();
+    Delay_Timer_Start();
+    
+    //Write Controlregdatatimer and controlregdealytimer value to 1 to keep them reset
+    Control_Reg_Data_Timer_Write(1);
+    Control_Reg_Delay_Timer_Write(1);
+    
+    //The pin value will be written as 1 as the starting condition
+    DHT_Pin_Write(1);
+    
+    //Initilize dhtpinisr and delaytimeriser
+    DHT_Pin_ISR_StartEx(DHT_Pin_ISR_Handler);
+    Delay_Timer_ISR_StartEx(Delay_Timer_ISR_Handler);
+    
+    //Disable the ISR compondents
+    DHT_Pin_ISR_Disable();
+    Delay_Timer_ISR_Disable();
+    
+    //Initilize count to 0 and executed to 1
+    count = 0;
+    executed = 1;
+}   
+
+
+//What dht_start does:
+//    Starts the dht components
+//    Get output from dht11
+//    Send out results
+//
+//Paramers: None
+//Returns: None
+
+
+void dht_start(){
+    
+    // Enable Global Interrupts
+    CyGlobalIntEnable;
```

Comment:
This gets done in the main function. In external modules, you can safely assume that global interrupts are enabled.